src/components/Requirements/SubRequirement.vue

```diff
@@ -233,28 +233,34 @@ export default Vue.extend({
       const allTakenCourseIds = new Set(this.subReq.courses.flat().map(course => course.courseId));
       const slots: SubReqCourseSlot[] = [];
 
-      const coursesTaken = this.subReq.courses;
-
-      coursesTaken.forEach((subReqCourseSlot, i) => {
-        if (subReqCourseSlot.length > 0) {
-          subReqCourseSlot.forEach(subReqCourse => {
-            slots.push({ isCompleted: true, courses: [subReqCourse] });
-          });
-          // Create new IncompletedSubReqCourse slot if all credits or courses not met
-          // but only one CompletedSubReqCourse slot exists
-          if (coursesTaken.length === 1 && !this.isCompleted) {
-            slots.push({
-              isCompleted: false,
-              courses: generateSubReqIncompleteCourses(allTakenCourseIds, subReqEligibleCourses[i]),
-            });
-          }
+      if (subReqSpec.fulfilledBy === 'credits') {
+        if (this.isCompleted) {
+          slots.push({ isCompleted: true, courses: this.subReq.courses[0] });
         } else {
           slots.push({
             isCompleted: false,
-            courses: generateSubReqIncompleteCourses(allTakenCourseIds, subReqEligibleCourses[i]),
+            courses: generateSubReqIncompleteCourses(allTakenCourseIds, subReqEligibleCourses[0]),
           });
         }
```

Comment:
By special casing `'credits'` like this, I think we lost the ability to generate a new IncompleteSubReqCourse slot for credits. Left is this branch's staging and right is dev staging.


https://user-images.githubusercontent.com/54298311/111172601-572df280-857c-11eb-9783-fae72c327056.mov